### PR TITLE
tests: provide unittests for gnrc_sixlowpan_frag/rbuf

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This provides tests for GNRC's 6LoWPAN reassembly buffer. The reason I put these unittests into their own application is that I intentionally wanted to test this module very isolated + it allows me to receive the reassembled packet without any interference of potentially initialized packets.

### Testing procedure
Run `make test` for `tests/gnrc_sixlowpan_frag` on any platform you prefer.

### Issues/PRs references
~~None yet, but I found a bug while testing, which requires fixing before we can merge this.~~

~~Requires #10852 to pass.~~ (merged)

### CODE REVIEW

File: sys/net/gnrc/network_layer/sixlowpan/frag/rbuf.c [hunk 1]
```diff
@@ -87,11 +87,16 @@ static int _rbuf_add(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
...
+    sixlowpan_frag_n_t *frag = pkt->data;
     rbuf_int_t *ptr;
     uint8_t *data = ((uint8_t *)pkt->data) + sizeof(sixlowpan_frag_t);
     size_t frag_size;
 
+    /* check if provided offset is the same as in fragment */
+    assert(((((frag->disp_size.u8[0] & SIXLOWPAN_FRAG_DISP_MASK) ==
+                SIXLOWPAN_FRAG_1_DISP)) && (offset == 0)) ||
+           ((((frag->disp_size.u8[0] & SIXLOWPAN_FRAG_DISP_MASK) ==
+                SIXLOWPAN_FRAG_N_DISP)) && (offset == (frag->offset * 8U))));
```
Comment (miri64) [inline sys/net/gnrc/network_layer/sixlowpan/frag/rbuf.c]: `gnrc_sixlowpan_frag` internally derives the offset value directly from the fragment header

https://github.com/RIOT-OS/RIOT/blob/1f7292c358ccda95090fb15ff29ac93d674f26e2/sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c#L306-L322

So for normal usage within GNRC this assertion is redundant, but to make the tests of `rbuf_add` 100% water-tide I added it in a separate commit (48d73191c5e6161ed1c2474004581b14d5a2062b when writing this comment).

**Comment by miri64** [comment]
Rebased and squashed to current master to include #10852.

**Comment by miri64** [comment]
Ping @aabadie? IMHO this should be merged before feature freeze so these tests are included into the release tests.

**Comment by aabadie** [comment]
> this should be merged before feature freeze so these tests are included into the release tests.

I'm not against it but I don't feel enough knowledgeable with the proposed changes.

I tried on iotlab-m3 and native and the new unittests are OK. Just one thing: when I remove the changes in the `sys/net/gnrc/network_layer/sixlowpan/frag/rbuf.h/c`, except the one related to TEST_SUITES, the unittests are still passing. So are these changes really required ?

**Comment by aabadie** [comment]
Indeed, I didn't change anything in `rbuf.h`.

> all these changes are in commits separate from the commit introducing the tests and I intent to keep them separate from them.

This is perfectly fine, it's just that I'm wondering if they are actually needed. I was expecting the tests to fail by removing some of the changes.

**Comment by miri64** [comment]
@aabadie may I squash then?

**Comment by aabadie** [comment]
> may I squash then?

yes go ahead

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
> I also pinged with large payloads on the testbed with success.

I asked @cgundogan offline about this because I was confused since the tests provided here don't use the radio. He did it (with `gnrc_networking`) because he was unsure if my changes to `gnrc_sixlowpan_frag` might break something and was assured that they don't with this.

**Comment by aabadie** [comment]
All green, let's go.

**Pull request merged**